modules/base/template.go

```diff
@@ -96,9 +96,42 @@ func ToUtf8(content string) string {
 	return res
 }
 
+// Replaces all prefixes 'old' in 's' with 'new'.
+func ReplaceLeft(s, old, new string) string {
+	old_len, new_len, i, n := len(old), len(new), 0, 0
+	for ; i < len(s) && strings.HasPrefix(s[i:], old); n += 1 {
+		i += old_len
+	}
+
+	// simple optimization
+	if n == 0 {
+		return s
+	}
+
+	// allocating space for the new string
+	newLen := n*new_len + len(s[i:])
+	replacement := make([]byte, newLen, newLen)
+
+	j := 0
+	for ; j < n*new_len; j += new_len {
+		copy(replacement[j:j+new_len], new)
+	}
+
+	copy(replacement[j:], s[i:])
+	return string(replacement)
+}
+
```

Comment:
I simplified this function as follows:

```go
func ReplaceLeft2(s, old, new string) string {
	old_len, i, n := len(old), 0, 0
	for ; i < len(s) && strings.HasPrefix(s[i:], old); n += 1 {
		i += old_len
	}

	// simple optimization
	if n == 0 {
		return s
	}

	return strings.Repeat(new, n) + s[i:]
}
```

And did simple benchmark:

```go
func Benchmark_ReplaceLeft(b *testing.B) {
	for n := 0; n < b.N; n++ {
		ReplaceLeft(`Change 1 did this. Change 2 did this.

These changes were very good. I'm pretty happy with them. Here they are:
    * Change 1
    * Change 2`, " ", "&nbsp;")
	}
}

func Benchmark_ReplaceLeft2(b *testing.B) {
	for n := 0; n < b.N; n++ {
		ReplaceLeft2(`Change 1 did this. Change 2 did this.

These changes were very good. I'm pretty happy with them. Here they are:
    * Change 1
    * Change 2`, " ", "&nbsp;")
	}
}
```

The performance are pretty close but mine one has less code:

```
Benchmark_ReplaceLeft-8 	200000000	         8.37 ns/op
Benchmark_ReplaceLeft2-8	200000000	         8.08 ns/op
```

How do you think?